Formula/opencoarrays.rb

```diff
@@ -0,0 +1,55 @@
+class Opencoarrays < Formula
+  desc "open-source coarray Fortran ABI, API, and compiler wrapper"
+  homepage "http://opencoarrays.org"
+  url "https://github.com/sourceryinstitute/opencoarrays/releases/download/1.8.2/OpenCoarrays-1.8.2.tar.gz"
+  sha256 "dc5b98a62de85e38500f431b05290415d4a4a87001c0ba03fbe57403ad62010b"
+
+  head "https://github.com/sourceryinstitute/opencoarrays.git"
+
+  option "without-test", "Skip build time tests (not recommended)"
+
+  depends_on "gcc"
+  depends_on :fortran
+  depends_on :mpi => :cc
+  depends_on "cmake" => :build
+
+  def install
+    fc_string = Utils.popen_read(ENV.fc, "--version").split("\n").first.chomp
+    ohai "Building OpenCoarrays with the following fortran compiler: \n" + fc_string
+    odie "OpenCoarrays requires GFortran" unless fc_string =~ /gnu/i
+    fc_version = fc_string.split(" ").last
+    odie "OpenCoarrays requires GFortran 5.4 or newer" unless Gem::Version.new(fc_version) >= Gem::Version.new("5.4.0")
```

Comment:
@MikeMcQuaid I'm not sure if this is the best/correct way to do this, but I have added checks on the Fortran compiler by parsing some of the output of `ENV.fc --version` I print diagnostic info (the first line of that output) with `ohai` and I throw errors if it's not `GNU Fortran` and also `>= 5.4.0` using `odie`.

Unless you want me to adjust any of this, I'm quite happy with it and think that we're finally good to go.